Travel Scheduler/View Controllers/HomeCollectionViewController.m

```diff
@@ -71,64 +75,50 @@ - (void)handleRefresh:(UIRefreshControl *)refreshControl {
-//            CGFloat green = arc4random() % 255;
-//            CGFloat blue = arc4random() % 255;
-//            UIColor *color = [UIColor colorWithRed:red/255.0 green:green/255.0 blue:blue/255.0 alpha:1.0f];
-//            [colorArray addObject:color];
-//        }
-//        [mutableArray addObject:colorArray];
-//    }
-//    self.colorArray = [NSArray arrayWithArray:mutableArray];
-//    self.contentOffsetDictionary = [NSMutableDictionary dictionary];
-//}
-
-
 #pragma mark - UITableViewDataSource Methods
 
-- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
-{
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
     return 1;
 }
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
-{
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
     return 3;
 }
 
-- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
-{
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
     static NSString *cellIdentifier = @"cellIdentifier";
     PlacesToVisitTableViewCell *cell = (PlacesToVisitTableViewCell *)[tableView dequeueReusableCellWithIdentifier:cellIdentifier];
     if (cell == nil){
         cell = [[PlacesToVisitTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
         CGRect myFrame = CGRectMake(10.0, 0.0, 220, 25.0);
         cell.labelWithSpecificPlaceToVisit = [[UILabel alloc] initWithFrame:myFrame];
+        cell.hub = self.hub;
+        cell.arrayOfPlaces = [[NSMutableArray alloc] init];
         if(indexPath.row == 0){
+            self.areWeInAttractions = YES;
+            self.areWeInLodging = NO;
+            self.areWeInRestaurant = NO;
             cell.labelWithSpecificPlaceToVisit.text = @"Attractions";
+            cell.typeOfPlaces = @"attractions";
+            NSMutableSet *set = [NSMutableSet setWithArray:self.hub.dictionaryOfArrayOfPlaces[@"museum"]];
+            [set addObjectsFromArray:self.hub.dictionaryOfArrayOfPlaces[@"park"]];
+            cell.arrayOfPlaces = (NSMutableArray *)[set allObjects];
         }
         else if (indexPath.row == 1) {
+            self.areWeInAttractions = NO;
+            self.areWeInLodging = NO;
+            self.areWeInRestaurant = YES;
             cell.labelWithSpecificPlaceToVisit.text = @"Restaurants";
+            cell.typeOfPlaces = @"restaurant";
+            cell.arrayOfPlaces = self.hub.dictionaryOfArrayOfPlaces[cell.typeOfPlaces];
         }
         else if (indexPath.row == 2){
+            self.areWeInAttractions = NO;
+            self.areWeInLodging = YES;
+            self.areWeInRestaurant = NO;
             cell.labelWithSpecificPlaceToVisit.text = @"Hotels";
+            cell.typeOfPlaces = @"lodging";
+            cell.arrayOfPlaces = self.hub.dictionaryOfArrayOfPlaces[cell.typeOfPlaces];
```

Comment:
I think you could set up all the cell properties (like label, type of place, array...) in the cell class so that it doesn't look so crowded here. Ex: have a method like setupProperties:(NSString)text, (NSString *)placeType, (NSArray *)array... in the cell class so you just have to call [cell setupProperties:...] here instead of having similar code in each if statement.

Reply:
I agree with Angela's comment. It seems the code repeats most stuff in else if conditions which is a great indication of a helper method. I think creating the method in the cell sounds good since code would be less lengthy in this file (which I think is good).

Reply:
Done! Thanks guys